src/equations/maxwell_1d.jl

```diff
@@ -0,0 +1,106 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    MaxwellEquations1D
+
+The Maxwell equations of electro dynamics
+```math
+\frac{\partial}{\partial t}
+\begin{pmatrix}
+E \\ B
+\end{pmatrix}
++ 
+\frac{\partial}{\partial x}
+\begin{pmatrix}
+c^2 B \\ E
+\end{pmatrix}
+=
+\begin{pmatrix}
+0 \\ 0 
+\end{pmatrix}
+```
+in one dimension with speed of light `c = 299792458 m/s` (in vacuum).
+In one dimension the Maxwell equations reduce to a wave equation.
+The orthogonal magnetic (e.g.`B_y`) and electric field (`E_z`) propagate as waves 
+through the domain in `x`-direction.
+For reference, see 
+- e.g. p.15 of Numerical Methods for Conservation Laws: From Analysis to Algorithms 
+  https://doi.org/10.1137/1.9781611975109
+
+- or equation (1) in https://inria.hal.science/hal-01720293/document
+"""
+struct MaxwellEquations1D{RealT <: Real} <:
+       AbstractLinearScalarAdvectionEquation{1, 2}
+    speed_of_light::SVector{1, RealT} # c
```

Comment:
Why do you store the speed of light as an `SVector` instead of a scalar?

Reply:
This is based on 

https://github.com/trixi-framework/Trixi.jl/blob/b4eef6d87d1e5a737ab31ad9c1e5bad293a19b4d/src/equations/linear_scalar_advection_1d.jl#L19

but I agree that this does not really make sense as we will probably not add directionally different sppeds of light on the main branch.